specification/under_development/profiles/Base_Profile.md

```diff
@@ -0,0 +1,484 @@
+the [labeling schema](#output-recording) used in the recorded output can be
+identified by looking at the metadata in the produced output. See the
+specification of the [output schemas](../output_schemas/) for more information
+about how metadata is represented in the output schema.
+
+Custom function attributes will show up as part of an [attribute
+group](https://releases.llvm.org/13.0.1/docs/LangRef.html#attrgrp) in the IR.
+Attribute groups are numbered such that they can be easily referenced by
+multiple function definitions or global variables. Consumers of Base Profile
+compliant programs must not rely on a particular numbering, but instead look for
+function to which an attribute with the name `"entry_point"` is attached to
+determine which one to invoke when the program is launched.
+
+Within the restrictions imposed by the Base Profile, the number of qubits that
+are needed to execute the quantum program must be known at compile time. This
+number is captured in the form of the `"required_qubits"` attribute attached to
+the entry point. The value of the attribute must be the string representation of
+a 64-bit integer constant.
+
+Similarly, the number of measurement results that need to be stored when
+executing the entry point function is captured by the `"required_results"`
+attribute. Since qubits cannot be used after measurement, in the case of the
+Base Profile, this value is equal to the number of measurement results in the
+program output.
+
+For a program to be valid, both the number of qubits used and the number of
+measurements to store must be strictly positive. More details on the usage of
+[qubits and results](#qubit-and-result-usage) in general can be found in the
+sections below.
+
+Beyond the entry point specific requirements related to attributes, custom
+attributes may optionally be attached to any of the declared functions.
+Furthermore, the following [LLVM
+attributes](https://llvm.org/docs/LangRef.html#function-attributes) may be used
+according to their intended purpose on function declarations and call sites:
+`inlinehint`, `nofree`, `norecurse`, `readnone`, `readonly`, `writeonly`, and
+`argmemonly`.
+
+### Function Body
+
+The function body consists of two blocks, connected by an unconditional
+branching into the second block at the end of the first block.
+
+The first block contains only calls to QIS functions. Any number of calls to QIS
+functions may be performed. To be compatible with the Base Profile these
+functions must return void. Any arguments to invoke them must be inlined into
+the call itself; they must be constants or a pointer representing a [qubit or
+result](#data-types-and-values) value. All function calls leading to a unitary
+transformation of the quantum state must precede any measurements; any calls
+that perform a measurement of one or more qubit(s) can only be followed by other
+such calls or the block terminating branching into the second block. The section
```

Comment:
"terminating" => "terminator"
Should we make it clear that it can only be an [unconditional `br` instruction](https://llvm.org/docs/LangRef.html#br-instruction) in the form of `br label block2_label`?

Reply:
Rephrased and clarified that it is an unconditional branch.